ekncontent/ekncontent/eknc-engine.c

```diff
@@ -0,0 +1,508 @@
+/* Copyright 2016 Endless Mobile, Inc. */
+
+#include "eknc-engine.h"
+
+#include "eknc-domain.h"
+#include "eknc-xapian-bridge.h"
+#include "eknc-utils.h"
+
+/**
+ * SECTION:engine
+ * @title: Engine
+ * @short_description: Grab content for a specific application
+ *
+ * The knowledge content engine is the main portal for query and fetching
+ * content for knowledge applications. Usually you will be using this to fetch
+ * content for a single application, in which case you should set the
+ * #EkncEngine:default-app-id property.
+ */
+struct _EkncEngine
+{
+  GObject parent_instance;
+
+  gchar *default_app_id;
+  gchar *language;
+  EkncXapianBridge *xapian_bridge;
+  GHashTable *domains;
```

Comment:
nitpick: Same as comment from last time about `GSList`, nice to have a comment here about the hash table's key and value types